### ISSUE TITLE
pre-trained model size & training settings?

### ISSUE THREAD

**Issue by StarsMyDestination** [zeliu98/CloserLook3D#4]
Hi, zeliu:
Thank you for your excellent work and open source code!
Since the paramter number is around 1M~20M, why the pretrained model has around 150M(for example posPool pytoch on ModelNet40)?
In my practice, 1M parameter should generate less than 10M model. 
This question has nothing to do with the code or paper, but I wonder is it normal?

thx!

**Comment by zeliu98** [comment]
Hi @StarsMyDestination . That's a good question. 
This is because we save more things than just the model parameters:
`
state = {
        'config': config,
        'model': model.state_dict(),
        'optimizer': optimizer.state_dict(),
        'scheduler': scheduler.state_dict(),
        'epoch': epoch,
        'best_acc': best_acc,
    }
` 

I measured the size of each part:
`
-rw-rw-r-- 1 liuze liuze   12 7月  30 14:56 best_acc.pkl
-rw-rw-r-- 1 liuze liuze 2.3K 7月  30 14:56 config.pkl
-rw-rw-r-- 1 liuze liuze    6 7月  30 14:56 epoch.pkl
-rw-rw-r-- 1 liuze liuze  75M 7月  30 14:56 model.pkl
-rw-rw-r-- 1 liuze liuze  75M 7月  30 14:56 optimizer.pkl
-rw-rw-r-- 1 liuze liuze 149M 7月  30 14:55 pospool_xyz_avg.pth
-rw-rw-r-- 1 liuze liuze 2.4K 7月  30 14:56 scheduler.pkl
`
It shows that the 'optimizer' takes a lot of size. Perhaps, we could remove them if necessary.

**Comment by StarsMyDestination** [comment]
I see, I should check the code first.

**Comment by StarsMyDestination** [comment]
Another question@zeliu98:
I re-trained a model(adaptiveweight_dp_fc1_avg, pytorch), the GPU consume is far more than i thougt(~18GB).
I am trying to reduce batch-size,  and i need some advice. 
Can you paste your training commands here?

thx!

**Comment by zeliu98** [comment]
> Another question @zeliu98:
> I re-trained a model(adaptiveweight_dp_fc1_avg, pytorch), the GPU consume is far more than i thougt(~18GB).
> I am trying to reduce batch-size, and i need some advice.
> Can you paste your training commands here?
> 
> thx!

That's strange. Here is the command:
```
python -m torch.distributed.launch --master_port 12346 --nproc_per_node 1  function/train_modelnet_dist.py --cfg cfgs/modelnet/pospool_xyz_avg.yaml
```
It takes about 7963MiB GPU memory.

**Comment by StarsMyDestination** [comment]
> > Another question @zeliu98:
> > I re-trained a model(adaptiveweight_dp_fc1_avg, pytorch), the GPU consume is far more than i thougt(~18GB).
> > I am trying to reduce batch-size, and i need some advice.
> > Can you paste your training commands here?
> > thx!
> 
> That's strange. Here is the command:
> 
> ```
> python -m torch.distributed.launch --master_port 12346 --nproc_per_node 1  function/train_modelnet_dist.py --cfg cfgs/modelnet/pospool_xyz_avg.yaml
> ```
> 
> It takes about 7963MiB GPU memory.

yes, when i run `pospool_xyz_avg.yaml`, it says 8156M(consistent with yours).
BUT when i run `pointwisemlp_dp_fi_df_fc1.yaml`, it comes to 26272M.

By the way, i use pytorch 1.5, cudnn 10.1.

**Comment by StarsMyDestination** [comment]
question about training time @zeliu98 ：
I use 1 GPU(bsz=16) training `pointwisemlp_dp_fi_df_fc1.yaml`, it costs 882 sec every epoch. SO, I need nearly 150h(6.25 days) to complete 600 epochs. Since you have done so much experiments, I don't think this is normal.
I use V100 doing above experiment.

**Comment by zeliu98** [comment]
Hi @StarsMyDestination. We do our experiments mainly on the `tensorflow` code base. As for `pytorch`, we use 4 GPUs for training and it takes about 2 days. The main reason for the slow speed is that we do subsampling and ball query in an online way using the `pt_custom_ops`. We are working on optimization for it. For now, there are some  possible solutions:

- using more GPUs for training
- increase the batch size if possible
- reduce the total epochs and change the `lr_decay_rate` correspondingly.
-  change the bottleneck ratio to a smaller number like `4`.
